drivers/ethernet/eth_w5500.c

```diff
@@ -567,10 +577,50 @@ static int w5500_init(const struct device *dev)
 			LOG_ERR("Unable to configure GPIO pin %u", config->reset.pin);
 			return -EINVAL;
 		}
-		gpio_pin_set_dt(&config->reset, 0);
+		gpio_pin_set_dt(&config->reset, 1);
 		k_usleep(500);
+		gpio_pin_set_dt(&config->reset, 0);
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(ready_gpios)
+		if (config->ready.port) {
+			if (!gpio_is_ready_dt(&config->ready)) {
+				LOG_ERR("GPIO port %s not ready", config->ready.port->name);
+				return -EINVAL;
+			}
+			if (gpio_pin_configure_dt(&config->ready, GPIO_INPUT)) {
+				LOG_ERR("Unable to configure GPIO pin %u", config->ready.pin);
+				return -EINVAL;
+			}
+
+			k_timepoint_t timeout = sys_timepoint_calc(WIZ550IO_RESET_DELAY);
+
+			while (!gpio_pin_get_dt(&config->ready)) {
+				if (sys_timepoint_expired(timeout)) {
+					LOG_ERR("W5500 not ready");
+					return -ETIMEDOUT;
+				}
+				k_msleep(1);
+			}
+		} else
+#endif
+		{
+			k_sleep(W5500_RESET_DELAY);
+		}
 	}
 
+#if !DT_INST_PROP(0, zephyr_random_mac_address) && !NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(0))
```

Comment:
The MAC address needs to be read out after a hardware reset and before software reset.

> Therefore, we recommend HW reset instead of SW reset. Nevertheless, if users want to use SW reset, we recommend to save MAC address ...... before SW reset, and writing those information to WIZ550io after SW reset.

https://docs.wiznet.io/Product/ioModule/wiz550io

Reply:
I understand. In that case, we should avoid soft reset IMO. Makes sense to move this to `w5500_set_macaddr` and `else` condition the `w5500_soft_reset` if ready gpio present?

Reply:
I am a little hesitant to remove the soft reset, as I'm not sure exactly what registers the PIC12F519 MCU configures in the W5500. Other than the mac address, registers should be configured by the Zephyr driver (or default/reset values).

I tested your suggestion and it does _seem_ to work.

Given the above, can you confirm whether you think this change should be made, and I will update accordingly

Reply:
Understood thanks